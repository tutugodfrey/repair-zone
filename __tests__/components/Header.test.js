import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ConnectedHeader, { Header, mapStateToProps }
  from '../../client/src/components/Header.jsx';

const mockStore = configureStore();
describe('<Header />', () => {
  describe('unconnected component test', () => {
    describe('admin dashboard', () => {
      let wrapper
      beforeEach(() => {
        const dispatch = jest.fn();
        const userData = {
          isAdmin: true,
          imgUrl: 'path/to/profile/photo',
        }
        wrapper = shallow(
        <Header
          dispatch={dispatch}
          userData={userData}
          idOfActiveTab={'view-requests'}
        />);
      });
      test('shoudld match the component snapshot for admin', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
      })
    });
    describe('user dashboard', () => {
      let wrapper
      beforeEach(() => {
        const dispatch = jest.fn();
        const userData = {
          isAdmin: false,
          imgUrl: 'path/to/profile/photo',
        }
        wrapper = shallow(
        <Header
          dispatch={dispatch}
          userData={userData}
          idOfActiveTab={'view-requests'}
        />);
      });
      test('shoudld match the component snapshot for user', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
      })
    })
  });

  describe('connected component test', () => {
    let wrapper;
    beforeEach(() => {
      const initialState = {
        userData: {
          isAdmin: false,
          imgUrl: 'path/to/profile/photo',
        }
      };
      const store = mockStore(initialState);
      wrapper = mount(
        <Provider store={store}>
          <ConnectedHeader
          />
        </Provider>
      );
    });
    test('should render with a div with #navbar-content', () => {
      const navbarContainer = wrapper.find('#navbar-content');
      expect(navbarContainer.length).toBe(1);
    });
    test('should render with a div with #navbar-content', () => {
      const navbarContainer = wrapper.instance();
    })
  });

  describe('mapStateToProps', () => {
    test('map the state of the component correctly', () => {
      const appState = {
        userData: {},
        idOfActiveTab: 'view request',
        someRandomField: {},
      };
      const expectedState = {
        userData: {},
        idOfActiveTab: 'view request',
      };
      const componentState = mapStateToProps(appState);
      expect(componentState).toEqual(expectedState);
    });
  });
});
