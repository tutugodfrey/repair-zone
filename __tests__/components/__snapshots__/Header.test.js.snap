// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> unconnected component test admin dashboard shoudld match the component snapshot for admin 1`] = `
<div>
  <div
    className="bg-success row"
    id="top-level-header"
  >
    <div
      className="col-5 offset-6 col-sm-4 offset-sm-8 col-md-3 offset-md-9"
    >
      <a
        className="ml-auto text-white py-1 px-2 border rounded"
        href="#"
        id="logout-link"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </div>
  <div
    className="row navbar navbar-expand-md navbar-light bg-success"
  >
    <div
      className="col-4 px-0 col-md-2"
      id="logo-div3"
    >
      <Link
        href="#"
        linkClass="logo text-white font-weight-bold"
        linkId="home-logo"
        linkText="Repair-Zone"
        onClick={[Function]}
      />
    </div>
    <NavButton
      aria-controls="narbar-content"
      aria-label="Toggle-navigation"
      buttonClass="navbar-toggler ml-auto"
      buttonId=""
      buttonName={
        <span
          className="navbar-toggler-icon"
        />
      }
      data-target="#navbar-content"
      data-toggle="collapse"
    />
    <div
      className="col-9 collapse navbar-collapse"
      id="navbar-content"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item m-1"
        >
          <Link
            href="#view-requests"
            linkClass="nav-link text-black py-1 border rounded"
            linkId="view-requests"
            linkText="View Request"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item m-1"
        >
          <Link
            href="#approved-request"
            linkClass="nav-link text-white py-1 border rounded"
            linkId="approved-requests"
            linkText="Approved Request"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item m-1"
        >
          <Link
            href="#resolved-request"
            linkClass="nav-link text-white py-1 border rounded"
            linkId="resolved-requests"
            linkText="Resolved Request"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item m-1"
        >
          <Link
            href="#contact-messages"
            linkClass="nav-link text-white py-1 border rounded"
            linkId="message"
            linkText="Contact Messages"
            onClick={[Function]}
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Header /> unconnected component test user dashboard shoudld match the component snapshot for user 1`] = `
<div>
  <div
    className="bg-success row"
    id="top-level-header"
  >
    <div
      className="col-5 offset-6 col-sm-4 offset-sm-8 col-md-3 offset-md-9"
    >
      <a
        className="ml-auto text-white py-1 px-2 border rounded"
        href="#"
        id="logout-link"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </div>
  <div
    className="row navbar navbar-expand-md navbar-light bg-success"
  >
    <div
      className="col-4 px-0 col-md-2"
      id="logo-div3"
    >
      <Link
        href="#"
        linkClass="logo text-white font-weight-bold"
        linkId="home-logo"
        linkText="Repair-Zone"
        onClick={[Function]}
      />
    </div>
    <NavButton
      aria-controls="narbar-content"
      aria-label="Toggle-navigation"
      buttonClass="navbar-toggler ml-auto"
      buttonId=""
      buttonName={
        <span
          className="navbar-toggler-icon"
        />
      }
      data-target="#navbar-content"
      data-toggle="collapse"
    />
    <div
      className="col-9 collapse navbar-collapse"
      id="navbar-content"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item m-1"
        >
          <Link
            href="#view-user-request"
            linkClass="nav-link text-black py-1 border active rounded"
            linkId="view-requests"
            linkText="View Request"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item m-1"
        >
          <Link
            href="#make-request"
            linkClass="nav-link text-white py-1 border inactive rounded"
            linkId="make-request"
            linkText="Make Request"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item m-1"
        >
          <Link
            href="#contact-messages"
            linkClass="nav-link text-white py-1 border inactive rounded"
            linkId="message"
            linkText="Contact Messages"
            onClick={[Function]}
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
